src/routes/reference/basic-reactivity/create-signal.mdx

```diff
@@ -15,147 +15,178 @@ description: >-
 
+type Setter<T> = {
+  <U extends T>(value: Exclude<U, Function> | ((prev: T) => U)): U;
+  <U extends T>(value: (prev: T) => U): U;
+  <U extends T>(value: Exclude<U, Function>): U;
+  <U extends T>(value: Exclude<U, Function> | ((prev: T) => U)): U;
+}
+
+interface SignalOptions<T> {
+  name?: string;
+  equals?: false | ((prev: T, next: T) => boolean);
+  internal?: boolean;
+}
 ```
 
-The Signal's value starts out equal to the passed first argument `initialValue` (or undefined if there are no arguments).
-The `createSignal` function returns a pair of functions as a two-element array: a getter (or accessor) and a setter.
-In typical use, you would destructure this array into a named Signal like so:
+## Parameters
 
-```tsx
-const [count, setCount] = createSignal(0)
-const [ready, setReady] = createSignal(false)
-```
+### `value`
 
-Calling the getter (e.g., `count()` or `ready()`) returns the current value of the Signal.
+- **Type:** `T` 
+- **Default:** `undefined`
 
-Crucial to automatic dependency tracking, calling the getter within a tracking scope causes the calling function to depend on this Signal, so that function will rerun if the Signal gets updated.
+The initial value for the signal. If no initial value is provided, the signal's type is automatically extended with `undefined`.
 
-Calling the setter (e.g., `setCount(nextCount)` or `setReady(nextReady)`) sets the Signal's value and updates the Signal (triggering dependents to rerun) if the value actually changed (see details below).
-The setter takes either the new value for the signal or a function that maps the previous value of the signal to a new value as its only argument.
-The updated value is also returned by the setter. As an example:
+### `options` 
 
-```tsx
-// read signal's current value, and
-// depend on signal if in a tracking scope
-// (but nonreactive outside of a tracking scope):
-const currentCount = count()
+- **Type:** `SignalOptions<T>` 
+- **Default:** `undefined`
 
-// or wrap any computation with a function,
-// and this function can be used in a tracking scope:
-const doubledCount = () => 2 * count()
+Configuration object for the signal.
 
-// or build a tracking scope and depend on signal:
-const countDisplay = <div>{count()}</div>
+#### `name`
 
-// write signal by providing a value:
-setReady(true)
+- **Type:** `string`
+- **Default:** `undefined`
 
-// write signal by providing a function setter:
-const newCount = setCount((prev) => prev + 1)
+A name for the signal used for debugging purposes in development mode.
 
-```
+#### `equals`
 
-:::note
-	If you want to store a function in a Signal you must use the function form:
+- **Type:** `false | ((prev: T, next: T) => boolean)`
+- **Default:** `false`
 
-	```tsx
-	setValue(() => myFunction);
-	```
+A custom comparison function to determine when the signal should update. 
+If set to `false`, the signal will always update regardless of value equality. By default, signals use reference equality (`===`).
```

Comment:
"This can be useful to create a Signal<void> that triggers manual updates in the reactive system." between those sentences.